src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -638,13 +639,22 @@ public void run() {
                     LOG.info("My election bind port: " + addr.toString());
                     setName(addr.toString());
                     ss.bind(addr);
+                    ss.setSoTimeout(10 * 1000); // Ten seconds
+                    long acceptStartTime = System.currentTimeMillis();
                     while (!shutdown) {
-                        client = ss.accept();
-                        setSockOpts(client);
-                        LOG.info("Received connection request "
-                                + client.getRemoteSocketAddress());
-                        receiveConnection(client);
-                        numRetries = 0;
+                        try {
+                            client = ss.accept();
+                            setSockOpts(client);
+                            LOG.info("Received connection request "
+                                     + client.getRemoteSocketAddress());
+                            receiveConnection(client);
+                            numRetries = 0;
+                        } catch (SocketTimeoutException e) {
+                            LOG.warn("The socket is listening for the election accepted "
+                                     + "an unexpected timeout ["
+                                     + (System.currentTimeMillis() - acceptStartTime) + "]milliseconds"
+                                     + "after the call to accept(). is this an instance of bug ZOOKEEPER-2836?");
```

Comment:
I don't love this error message and it doesn't make sense because we've set the socket timeout above so it will never time out based on that weird possible JVM error. So either we just continue after a timeout, or leave the timeout at 0 and leave a log statement that indicates it timed out unexpectedly but will retry, but the double fix doesn't really make sense to me.